### DISCUSSION TITLE
'Error: Cannot read properties of undefined' in derivative-field

### DISCUSSION THREAD

**Discussion by webwurm** [AntlerVC/firetable#1053] [Support / Q&A] [answered]
Lately I get very often the error "`Cannot read property of undefined (reading 'path')`' when doing an **"Evaluate all"** in a derivative column. When I do all of the rows one by one it works.

My latest function was a string-field out of a reference-field (which I needed for FlutterFlow):
```
const derivative:Derivative = async ({row,ref,db,storage,auth,logging})=>{
  // Generate a String-field with the UserID-ref., for use in FF
  return (row.uid_user==null) ? null : row.uid_user.id;
}
```
Again: If I do it for each row, it works - if I do "Evaluate all" I get the error above.

Am I doing something wrong or did something change in Rowy? Thanks!

**Comment by bofeiw** [comment, marked as answer]
Thanks for spotting this bug. I have created a fix and it will be released soon. 
https://github.com/rowyio/backend/pull/24
